tools/assets-automation/assets-maintenance-tool/Azure.Sdk.Tools.Assets.MaintenanceTool/Scan/AssetsScanner.cs

```diff
@@ -0,0 +1,359 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using Azure.Sdk.Tools.Assets.MaintenanceTool.Model;
+using Azure.Sdk.Tools.TestProxy.Common;
+using Azure.Sdk.Tools.TestProxy.Common.Exceptions;
+using Azure.Sdk.Tools.TestProxy.Store;
+using Microsoft.Extensions.FileSystemGlobbing;
+
+namespace Azure.Sdk.Tools.Assets.MaintenanceTool.Scan
+{
+    /// <summary>
+    /// Used to walk through repo configurations and locate all assets.
+    /// </summary>
+    public class AssetsScanner
+    {
+        public string WorkingDirectory { get; set; }
+        public static readonly string GIT_TOKEN_ENV_VAR = "GIT_TOKEN";
+
+        private string ResultsFile {
+            get {
+                return Path.Combine(WorkingDirectory, "output.json");
+            }
+        }
+
+        public GitProcessHandler handler { get; set; } = new GitProcessHandler();
+
+        public AssetsScanner(string workingDirectory)
+        {
+            WorkingDirectory = workingDirectory;
+        }
+
+        public AssetsScanner()
+        {
+            WorkingDirectory = Directory.GetCurrentDirectory();
+        }
+
+        /// <summary>
+        /// Walk a run configuration an create a resultSet of all found assets.json references.
+        /// 
+        /// This function automatically takes previous output into account by checking in the current
+        /// working directory for an "output.json" file.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <returns></returns>
+        public AssetsResultSet Scan(RunConfiguration config)
+        {
+            var resultSet = new List<AssetsResult>();
+            var existingResults = ParseExistingResults();
```

Comment:
For me, having `AssetsResultSet? existingResultSet` instead of `var existingResults` would help a little bit in comprehending the code.

In general, I dislike `var` unless the type is obvious (usually because it is on the right hand side) or dealing with humongous LINQ types.

Reply:
Yep it's not immediately apparent on the right side.